.github/workflows/test-build.yml

```diff
@@ -0,0 +1,37 @@
+# This workflow will do a clean installation of node dependencies, cache/restore them, build the source code and run tests across different versions of node
+# For more information see: https://docs.github.com/en/actions/automating-builds-and-tests/building-and-testing-nodejs
+
+
+name: Build for Test Environment
+
+on:
+  push:

```

Comment:
I dont know that we want this to run on push, because we constantly push changes, and really want this on pull_request only, that way it only runs once a PR is opened